### ISSUE TITLE
Package effort-sharing code for easier reusability

### ISSUE THREAD

**Issue by chantalwuer** [imagepbl/effort-sharing#105]
Currently the classes for allocation etc. just sit in the top directory. This only allows (easy) import of the classes to files that are in the same level. 
To make sure the modules can be imported from other folders within the repo, we need to package the code. It is quite simple: we create a new folder e.g. "EffortSharingClasses" or Tools (or we can also break it down into "Allocation" "Datareading" etc.). Tbd. This is where we move all the classes.
Every folder needs an empty `__init__.py` file for it to be recognized as a package, plus a `main.py` in the first repo level. Once that is done, we can import classes like `from EffortSharingTools.class_allocation import allocation`. 

This step becomes necessary when we want to build unit testing from a `tests` folder and it also structures the repo better.

It might mean that we need to change the way we import the modules in the existing code. But it shouldn't be too much work with find and replace. 

We can discuss in January what you think about this @MarkMDekker.

**Comment by Peter9192** [comment]
In #111 I introduced a python package structure as suggested above. I also moved notebooks to a dedicated folder and added a `pyproject.toml` file. The resulting structure (omitting unrelated files) is:

```
.
├── notebooks
│   ├── Aggregator.ipynb
│   ├── DataExporters
│   ├── EffortSharing_Netherlands
│   ├── Main.ipynb
│   ├── Plotting
│   └── PostProcessing
├── pyproject.toml
└── src
    └── effortsharing
        ├── __init__.py
        ├── class_allocation.py
        ├── class_allocation_combinedapproaches.py
        ├── class_datareading.py
        ├── class_exports.py
        ├── class_policyscens.py
        ├── class_tempalign.py
        └── class_variancedecomp.py

```

The top-level pyproject.toml file turns the code in `src/effortsharing` into a pip-installable package that you can easily import anywhere regardless of the path, as long as the environment is activated. Note that `main.py` is not really necessary is this case. 

This structure is inspired by https://cookiecutter-data-science.drivendata.org/, but with minimal adaptations from the current structure.

I will replay these changes in a separate PR.
